### DISCUSSION TITLE
Modal should support draggable

### DISCUSSION THREAD

**Discussion by ykant** [mantinedev/mantine#6952] [Feature requests]
Mantine is an awesome UI component library. Recently our team ready to start using Mantine to be our base components, but I found that it doesn't support draggable. 
Then i try to  make it work out like this

`
<Draggable disabled={!innerDraggable} handle=".modal-header">
        <Modal.Content>
          <Modal.Header
            className={classNames({
              [styles.modalHeader]: innerDraggable,
              'modal-header': innerDraggable,
            })}
          >
            <Modal.Title>{title}</Modal.Title>
            {withCloseButton && <Modal.CloseButton />}
          </Modal.Header>
          <Modal.Body>{children}</Modal.Body>
        </Modal.Content>
</Draggable>
`
It's great, but it's not work when I add a prop `bounds="parent"` to `Draggable`.
So I want to get help to make it better to use, any suggestion? Thanks for the suggestion.

**Comment by rtivital** [comment]
Why do you think that it is an issue with Mantine and not with react-draggable? Last time I've checked it, `bounds="parent"` did not work correctly in react-draggable.

**Comment by ykant** [comment]
![image](https://github.com/user-attachments/assets/ee7185b3-a5d8-41f4-ba2b-d28b864fe895)
![image](https://github.com/user-attachments/assets/a598a39e-2868-4633-b0a5-9219357f2031)
Because  the layout structure of `Modal.Content` 
1. `mantine-Modal-inner` element is fixed position and full of the window
2. `mantine-Modal-content` element is the flex item
3.  `Draggable` effects on the `mantine-Modal-inner`, so it's not work correctly
In fact I want to `Draggable` dragging the `mantine-Modal-content`, not the `mantine-Modal-inner`, but `Modal.Content` boxes them in a components that I can't insert  `Draggable`  between them.

**Comment by rhyek** [comment]
I had this working with react 18 like so:
```tsx
export function AppModal({
  children,
  title,
  ...rest
}: { children: React.ReactNode } & ModalProps) {
  return (
    <Modal.Root {...rest}>
      <Draggable handle=".mantine-Modal-header">
        <Modal.Content>
          <Modal.Header style={{ cursor: 'move' }}>
            <Modal.Title style={{ userSelect: 'none' }}>{title}</Modal.Title>
            <Modal.CloseButton />
          </Modal.Header>
          <Modal.Body>{children}</Modal.Body>
        </Modal.Content>
      </Draggable>
    </Modal.Root>
  );
}
```
This [doesn't work](https://github.com/react-grid-layout/react-draggable/issues/771) anymore in react 19. Per the comments in that issue we should be able to make it work so long as we can get a ref to the dom element for `.mantine-Modal-header`. Any ideas how I might do that? Thanks.

**Reply by rhyek** [reply]
So I was trying to get a ref to the dom element using `ref` on `Modal.Header`, but was being confused by an [error](https://github.com/react-grid-layout/react-draggable/issues/779) happening with react-draggable. However, once solving that it's pretty straightforward:

```tsx
export function AppModal({
  children,
  title,
  ...rest
}: { children?: React.ReactNode } & ModalProps) {
  // https://github.com/react-grid-layout/react-draggable/issues/779
  const nodeRef = useRef<HTMLElement>({} as any);
  return (
    <Modal.Root {...rest}>
      <Draggable
        nodeRef={nodeRef}
        // https://github.com/react-grid-layout/react-draggable/issues/761
        offsetParent={
          typeof document !== 'undefined' ? document.body : undefined
        }
        onStart={(event) => {
          if (
            event.target instanceof HTMLElement &&
            (() => {
              let element: HTMLElement | null = event.target;
              let depth = 0;
              while (element && depth < 10) {
                if (element.classList.contains('mantine-Modal-header')) {
                  return true;
                }
                element = element.parentElement;
                depth++;
              }
              return false;
            })()
          ) {
            return;
          }
          return false;
        }}
      >
        <Modal.Content>
          <Modal.Header style={{ cursor: 'move' }} ref={nodeRef}>
            <Modal.Title style={{ userSelect: 'none' }}>{title}</Modal.Title>
            <Modal.CloseButton />
          </Modal.Header>
          <Modal.Body>{children}</Modal.Body>
        </Modal.Content>
      </Draggable>
    </Modal.Root>
  );
}
```

**Reply by ykant** [reply]
```
<Draggable offsetParent={document.body} handle={`.${styles.darg}`} nodeRef={nodeRef as React.RefObject<HTMLDivElement>}>
          <MantineModal.Content ref={nodeRef}>
           ...
          </MantineModal.Content>
</Draggable>
```

try this
